service-providers/ip-packet-router/src/clients/connected_client_handler.rs

```diff
@@ -141,16 +130,19 @@ impl ConnectedClientHandler {
                     log::info!("client handler stopping: activity timeout: {}", self.sent_by);
                     break;
                 },
-                packets = self.encoder.buffer_timeout() => match packets {
-                    Some(packets) => {
-                        if let Err(err) = self.handle_buffer_timeout(packets).await {
-                            log::error!("client handler: failed to handle buffer timeout: {err}");
-                        }
-                    },
-                    None => log::trace!("no packets to send"),
+                _ = self.payload_topup_interval.tick() => {
+                    // Send an empty packet to trigger the buffer timeout
+                    if let Err(err) = self.handle_packet(Vec::new()).await {
```

Comment:
are we sure this won't result in client actually sending empty messages over the mixnet (and getting those warnings with "empty payload" on the other side?)

Reply:
Oops, that should be a `IprPacket::Flush` and not an empty packet